sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/ReturnMinimalResponseTest.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ *
+ */
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.CosmosItemProperties;
+import com.azure.cosmos.models.CosmosContainerProperties;
+import com.azure.cosmos.models.CosmosContainerRequestOptions;
+import com.azure.cosmos.models.CosmosContainerResponse;
+import com.azure.cosmos.models.CosmosDatabaseProperties;
+import com.azure.cosmos.models.CosmosDatabaseRequestOptions;
+import com.azure.cosmos.models.CosmosDatabaseResponse;
+import com.azure.cosmos.models.CosmosItemRequestOptions;
+import com.azure.cosmos.models.CosmosItemResponse;
+import com.azure.cosmos.models.IndexingMode;
+import com.azure.cosmos.models.IndexingPolicy;
+import com.azure.cosmos.models.ModelBridgeInternal;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.models.PartitionKeyDefinition;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ReturnMinimalResponseTest extends TestSuiteBase {
+
+    private final String preExistingDatabaseId = CosmosDatabaseForTest.generateId();
+    private final List<String> databases = new ArrayList<>();
+    private CosmosClient client;
+    private CosmosContainer container;
+    private CosmosDatabase createdDatabase;
+
+    //  Currently Gateway and Direct TCP support minimal response feature.
+    @Factory(dataProvider = "clientBuildersWithDirectTcpWithReturnMinimalResponse")
+    public ReturnMinimalResponseTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = {"simple"}, timeOut = SETUP_TIMEOUT)
+    public void beforeClass() {
+        assertThat(this.client).isNull();
+        this.client = getClientBuilder().buildClient();
+        createdDatabase = createSyncDatabase(client, preExistingDatabaseId);
+        CosmosAsyncContainer asyncContainer = getSharedMultiPartitionCosmosContainer(this.client.asyncClient());
+        container = client.getDatabase(asyncContainer.getDatabase().getId()).getContainer(asyncContainer.getId());
+    }
+
+    @AfterClass(groups = {"simple"}, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        safeDeleteSyncDatabase(createdDatabase);
+        for (String dbId : databases) {
+            safeDeleteSyncDatabase(client.getDatabase(dbId));
+        }
+        safeCloseSyncClient(client);
+    }
+
+    @Test(groups = {"simple"}, timeOut = TIMEOUT)
+    public void createDatabase_withMinimalResponseConfig() throws CosmosClientException {
+        CosmosDatabaseProperties databaseDefinition = new CosmosDatabaseProperties(CosmosDatabaseForTest.generateId());
```

Comment:
we usually target emulator for database and container CRUD test, because otherwise we may run out of master resource budget in prod.

Reply:
Yes, makes sense, will do that.